### DISCUSSION TITLE
Is there any way to increase or decrase handle count based on node size?

### DISCUSSION THREAD

**Discussion by Distrect** [xyflow/xyflow#3192] [Q&A] [answered]
In a custom node i would like to increase or decrase handle count on resize. It can be done(i know i answered my own question but please continıe to read) but what should i do if there is already connected edges ro that node? Is there any built in module to handles that or should be done custom? Thank you for your answers.

**Comment by bcakmakoglu** [comment, marked as answer]
[According to the docs.](https://reactflow.dev/docs/guides/custom-nodes/#using-multiple-handles)

>Note that if you are programmatically changing the position or number of handles in your custom node, you will need to use the [useUpdateNodeInternals](https://reactflow.dev/docs/api/hooks/use-update-node-internals/) hook to properly notify ReactFlow of changes. From here you should be able to build your custom nodes. In most cases we recommend to use custom nodes only. The built-in ones are just basic examples. You can find a list of the passed props and more information in the [custom node API section](https://reactflow.dev/docs/api/nodes/custom-nodes/).
